### PR TITLE
Fix safe-landing protocol for sparse domains

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
The protocol wasn't handling cases where no entities are nearby very well.

### CODE REVIEW

**Comment by SimonWalton-HiFi** [comment]
[simple-domain-content.zip](https://github.com/highfidelity/hifi/files/2312132/simple-domain-content.zip)

**Comment by SimonWalton-HiFi** [comment]
Original PR:
https://github.com/highfidelity/hifi/pull/13812

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12313/HighFidelity-Beta-PR13861-dd098f0.apk

**Comment by SimonWalton-HiFi** [comment]
## Test Plan

- Run DS and ACs from the PR.
- Upload the simple-domain-content above to the domain server by uploading the models.json to http://localhost:40100/content/#upload_content_group .
- Run interface from the PR; go to localhost/50.6162,51.6495,54.5718 . Repeat multiple times.

Before this PR the content of the domain may not load. After this PR the content (a box; also a tree model if uploaded to the asset server) should always load.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12313/HighFidelity-Beta-PR13861-dd098f0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12313/HighFidelity-Beta-Interface-PR13861-dd098f0.dmg

**Comment by antoninahifi** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12313/HighFidelity-Beta-PR13861-dd098f0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12313/HighFidelity-Beta-Interface-PR13861-dd098f0.exe

File: libraries/networking/src/udt/PacketHeaders.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ PacketVersion versionForPacketType(PacketType packetType) {
         case PacketType::AvatarIdentityRequest:
             return 22;
         default:
-            return 22;
+            return 23;
```
Comment (birarda) [inline libraries/networking/src/udt/PacketHeaders.cpp]: We should specifically raise `EntityQueryInitialResultsComplete`

**Comment by conklin94122** [comment]
Removed QA Ready. Please relabel when requested dev changes have been made.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12321/HighFidelity-Beta-PR13861-c4b916a.apk

File: libraries/networking/src/udt/PacketHeaders.cpp (hunk 1 above)
Comment (SimonWalton-HiFi) [reply libraries/networking/src/udt/PacketHeaders.cpp]: Should be good now.

**Comment by SimonWalton-HiFi** [comment]
[simple-andrew-scene.json.gz](https://github.com/highfidelity/hifi/files/2312342/simple-andrew-scene.json.gz)

Adding Andrew's scene for QA. I haven't been able to repro the bug with logging out/in.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12323/HighFidelity-Beta-PR13861-c0966d7.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12323/HighFidelity-Beta-PR13861-c0966d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12323/HighFidelity-Beta-Interface-PR13861-c0966d7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12323/HighFidelity-Beta-PR13861-c0966d7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12323/HighFidelity-Beta-Interface-PR13861-c0966d7.exe

**Comment by SimonWalton-HiFi** [comment]
Sorry - I've never really understood the logic behind the protocol versioning.

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by antoninahifi** [comment]
PASS @Conklin94122

**Pull request merged**